prov/sockets/src/sock_fabric.c

```diff
@@ -634,6 +622,62 @@ int sock_node_getinfo(const char *node, const char *service, uint64_t flags,
 	return ret;
 }
 
+static int sock_match_src_addr(struct slist_entry *entry, const void *src_addr)
+{
+	struct sock_host_list_entry *host_entry;
+	host_entry = container_of(entry, struct sock_host_list_entry, entry);
+
+        return (strcmp(host_entry->hostname, (char *) src_addr) == 0);
+}
+
+static int sock_addr_matches_interface(struct slist addr_list, struct sockaddr_in *src_addr)
```

Comment:
Is there a need for keeping the argument as struct? Cant we use pointer to list here?